core/src/main/java/tech/tablesaw/aggregate/CrossTab.java

```diff
@@ -144,13 +144,13 @@ private static Table rowPercents(Table xTabCounts) {
         }
 
         for (int i = 0; i < xTabCounts.rowCount(); i++) {
-            float rowTotal = (float) xTabCounts.numberColumn(xTabCounts.columnCount() - 1).get(i);
+            double rowTotal = xTabCounts.numberColumn(xTabCounts.columnCount() - 1).getDouble(i);
 
             for (int c = 1; c < xTabCounts.columnCount(); c++) {
                 if (rowTotal == 0) {
                     pctTable.numberColumn(c).append(Float.NaN);
                 } else {
-                    pctTable.numberColumn(c).append((float) xTabCounts.numberColumn(c).get(i) / rowTotal);
+                    pctTable.numberColumn(c).append(xTabCounts.numberColumn(c).getDouble(i) / rowTotal);
```

Comment:
alternatively you could do `doubleColumn(c).get(i)` which would be a bit more concise